renku/core/incubation/graph.py

```diff
@@ -63,72 +60,19 @@
 def generate_graph():
     """Return a command for generating the graph."""
     command = Command().command(_generate_graph).lock_project().require_migration()
-    return command.with_database(write=True, create=True).with_commit(commit_only=GRAPH_METADATA_PATHS)
+    return command.with_commit(commit_only=GRAPH_METADATA_PATHS)
 
 
-@inject.autoparams()
-def _generate_graph(client: LocalClient, database: Database, force=False):
+@inject.autoparams("client")
+def _generate_graph(client: LocalClient, force=False):
     """Generate graph and dataset provenance metadata."""
+    from renku.core.management.migrations.m_0009__new_metadata_storage import generate_new_metadata
 
-    def process_workflows(commit: Commit):
-        for file_ in commit.diff(commit.parents or NULL_TREE, paths=f"{client.workflow_path}/*.yaml"):
-            # Ignore deleted files (they appear as ADDED in this backwards diff)
-            if file_.change_type == "A":
-                continue
-
-            path: str = git_unicode_unescape(file_.a_path)
-
-            if not path.startswith(".renku/workflow") or not path.endswith(".yaml"):
-                continue
-
-            if not (client.path / path).exists():
-                communication.warn(f"Workflow file does not exists: '{path}'")
-                continue
-
-            workflow = Activity.from_yaml(path=path, client=client)
-            client.update_graphs(workflow)
-
-    def process_datasets(commit):
-        files_diff = list(commit.diff(commit.parents or NULL_TREE, paths=".renku/datasets/*/*.yml"))
-        paths = [git_unicode_unescape(f.a_path) for f in files_diff]
-        paths = [p for p in paths if len(Path(p).parents) == 4]  # Exclude files that are not in the right place
-        deleted_paths = [git_unicode_unescape(f.a_path) for f in files_diff if f.change_type == "A"]
-
-        datasets, deleted_datasets = _fetch_datasets(commit.hexsha, paths=paths, deleted_paths=deleted_paths)
-
-        revision = commit.hexsha
-        date = commit.authored_datetime
-
-        for dataset in datasets:
-            datasets_provenance.add_or_update(dataset, revision=revision, date=date)
-        for dataset in deleted_datasets:
-            datasets_provenance.remove(dataset, revision=revision, date=date, client=client)
-
-    commits = list(client.repo.iter_commits(paths=[f"{client.workflow_path}/*.yaml", ".renku/datasets/*/*.yml"]))
-    n_commits = len(commits)
-    commits = reversed(commits)
-
-    if force:
-        client.remove_graph_files()
-    elif client.has_graph_files():
+    try:
+        generate_new_metadata(client=client, force=force, remove=False)
+    except errors.OperationError:
```

Comment:
We will remove `renku graph generate` in future and always force-generate so this won't be an issue. The verbose message does not make sense in `generate_new_metadata` since users don't interact with it directly and `--force` won't have a meaning there.